src/documentalist.ts

```diff
@@ -87,9 +155,20 @@ export class Documentalist implements IApi {
     }
 
     public documentFiles(files: IFile[]) {
-        const documentation = {} as IDocumentalistData;
+        const documentation = {} as T;
         for (const { pattern, plugin } of this.plugins) {
-            documentation[plugin.name] = plugin.compile(this, files.filter((f) => pattern.test(f.path)));
+            const pluginDocumentation = plugin.compile(this, files.filter((f) => pattern.test(f.path)));
+            for (var key in pluginDocumentation) {
+                if (pluginDocumentation.hasOwnProperty(key)) {
+                    if (documentation.hasOwnProperty(key)) {
+                        console.warn(`
+                            WARNING: Duplicate plugin key "${key}".
+                            Your plugins are overwriting each other.
```

Comment:
maybe we can figure out a way to handle this. what if plugins could build on each other, like a pipeline?